minion/frontend/views.py

```diff
@@ -392,8 +393,19 @@ def robots():
 def api_session():
     return jsonify(success=True, data={'email': session['email'], 'role': session['role']})
 
+@app.route("/api/login", methods=["GET"])
+def api_login():
+    return jsonify(success=True, data={'login_type': config['login_conf']['login_type']})
+
 @app.route("/api/login", methods=["POST"])
-def persona_login():
+def login():
+    if config['login_conf']['login_type'] == 'persona':
+        return persona_login(request)
+
+    elif config['login_conf']['login_type'] == 'ldap':
+        return ldap_login(request)
+
```

Comment:
I'd rather this be a function lookup (getattr + callable) using config['login']['type'] + '_login', so that it can expand to any number of authentication schemes without having to have a pile of elif statements.